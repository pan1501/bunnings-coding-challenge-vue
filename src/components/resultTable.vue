<style scoped>
h3,
table {
  color: rgb(217, 217, 217);
}

h3 {
  text-align: center;
}

td,
th {
  padding: 8px;
}
.resultTable .source {
  text-align: center;
}
</style>
<template>
  <div class="resultTable">
    <h3>{{ this.title }}</h3>
    <table>
      <tr>
        <th v-bind:key="header" v-for="header in headers">
          {{ header }}
        </th>
      </tr>
      <tr v-bind:key="index" v-for="(row, index) in body">
        <td>{{ row.SKU }}</td>
        <td>{{ row.Description }}</td>
        <td class="source">{{ row.Source }}</td>
      </tr>
    </table>
  </div>
</template>

<script>
export default {
  name: "resultTable",
  props: {
    title: String,
    headers: Array,
    body: Array,
  },
};
</script>
