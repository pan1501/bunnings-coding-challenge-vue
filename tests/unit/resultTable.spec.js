import resultTable from "@/components/resultTable.vue";
import { mount } from "@vue/test-utils";

describe("resultTable tests", () => {
  describe("with correct data", () => {
    it("renders the correct title", () => {
      const title = "Expected Output";
      const wrapper = mount(<resultTable title={title} />);

      expect(wrapper.html()).toContain("<h3>Expected Output</h3>");
    });

    it("renders the correct headers", () => {
      const headers = ["SKU", "Description", "Source"];
      const wrapper = mount(<resultTable headers={headers} />);

      expect(wrapper.html()).toMatchSnapshot();
    });
    it("renders the correct body", () => {
      const body = [
        {
          SKU: "165-rcy-650",
          Description: "Tea - Decaf 1 Cup",
          Source: "A",
        },
        {
          SKU: "167-eol-949",
          Description: "Cheese - Grana Padano",
          Source: "A",
        },
      ];
      const wrapper = mount(<resultTable body={body} />);

      expect(wrapper.html()).toMatchSnapshot();
    });
  });
  describe("with empty data", () => {
    it("renders empty title", () => {
      const title = undefined;
      const wrapper = mount(<resultTable title={title} />);

      expect(wrapper.html()).toMatchSnapshot();
    });

    it("renders empty headers", () => {
      const headers = undefined;
      const wrapper = mount(<resultTable headers={headers} />);

      expect(wrapper.html()).toMatchSnapshot();
    });

    it("renders empty body", () => {
      const body = undefined;
      const wrapper = mount(<resultTable body={body} />);

      expect(wrapper.html()).toMatchSnapshot();
    });
  });
});
