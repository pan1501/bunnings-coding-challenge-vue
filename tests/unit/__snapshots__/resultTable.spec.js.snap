// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`resultTable tests with correct data renders the correct body 1`] = `
<div class="resultTable">
  <h3></h3>
  <table>
    <tr></tr>
    <tr>
      <td>165-rcy-650</td>
      <td>Tea - Decaf 1 Cup</td>
      <td class="source">A</td>
    </tr>
    <tr>
      <td>167-eol-949</td>
      <td>Cheese - Grana Padano</td>
      <td class="source">A</td>
    </tr>
  </table>
</div>
`;

exports[`resultTable tests with correct data renders the correct headers 1`] = `
<div class="resultTable">
  <h3></h3>
  <table>
    <tr>
      <th>SKU</th>
      <th>Description</th>
      <th>Source</th>
    </tr>
  </table>
</div>
`;

exports[`resultTable tests with empty data renders empty body 1`] = `
<div class="resultTable">
  <h3></h3>
  <table>
    <tr></tr>
  </table>
</div>
`;

exports[`resultTable tests with empty data renders empty headers 1`] = `
<div class="resultTable">
  <h3></h3>
  <table>
    <tr></tr>
  </table>
</div>
`;

exports[`resultTable tests with empty data renders empty title 1`] = `
<div class="resultTable">
  <h3></h3>
  <table>
    <tr></tr>
  </table>
</div>
`;
